pkgs/by-name/pr/protonmail-bridge-gui/package.nix

```diff
@@ -0,0 +1,105 @@
+{ lib
+, stdenv
+, pkg-config
+, libsecret
+, cmake
+, ninja
+, qt6
+, grpc
+, protobuf
+, zlib
+, gtest
+, sentry-native
+, protonmail-bridge
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "protonmail-bridge-gui";
+
+  inherit (protonmail-bridge) version src;
+
+  patches = [
+    ./use-nix-googletest.patch
+    ./designator-order-fix.patch
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DBRIDGE_APP_FULL_NAME=ProtonMailBridge" # Should be 'Proton Mail Bridge' but https://github.com/NixOS/nixpkgs/issues/114044
+    "-DBRIDGE_VENDOR=ProtonAG" # Should also be 'Proton AG'
+    "-DBRIDGE_REVISION=${finalAttrs.src.rev}"
+    "-DBRIDGE_TAG=v${finalAttrs.version}"
+    "-DBRIDGE_BUILD_ENV=dev"
+    "-DBRIDGE_APP_VERSION=${finalAttrs.version}"
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    ninja
+    qt6.qtbase
+    qt6.qtdeclarative
+    qt6.qtwayland
+    qt6.qtsvg
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    libsecret
+    grpc
+    protobuf
+    zlib
+    gtest
+    sentry-native
+  ];
+
+  sourceRoot = "source/internal/frontend/bridge-gui";
```

Comment:
I think, but I may be wrong, that you need to use something else.

Something like:

```nix
sourceRoot = "${src.name}/internal/frontend/bridge-gui";
```

Reply:
Thanks, since I'm using `finalAttrs` it would actually be

```nix
sourceRoot = "${finalAttrs.src.name}/internal/frontend/bridge-gui";
```

The derivation so far worked because `${finalAttrs.src.name}` happened to be `"source"`. Using `"source/internal/frontend/bridge-gui"` is simpler and more readable, but I'm happy to use the safer and more future-proof (CLI's `src.name` might change) `${finalAttrs.src.name}`.